File: .vscode/linked_list/delete_node.py
# Node class
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Function to delete the x-th node
def deleteNode(head, x):
    if head is None:
        return None

    # Deleting the first node
    if x == 1:
        return head.next

    current = head
    # Traverse to the (x-1)-th node
    for i in range(1, x - 1):
        if current is None or current.next is None:
            return head  # Position out of bounds
        current = current.next

    # Delete x-th node if exists
    if current.next is not None:
        current.next = current.next.next

    return head

# Function to print the linked list
def printList(head):
    while head:
        print(head.data, end=' ')
        head = head.next
    print()

# Function to build the linked list from a list of values
def buildLinkedList(values):
    if not values:
        return None
    head = Node(values[0])
    current = head
    for val in values[1:]:
        current.next = Node(val)
        current = current.next
    return head

# Main program
if __name__ == "__main__":
    # Input: list elements
    elements = list(map(int, input("Enter elements of linked list: ").split()))
    head = buildLinkedList(elements)

    # Input: position to delete
    x = int(input("Enter position to delete (1-based): "))

    # Delete node
    head = deleteNode(head, x)

    # Output: updated list
    print("Updated linked list:")
    printList(head)

File: linked_list/linked_list/delete_node.c
#include <stdio.h>
#include <stdlib.h>

// Node structure
struct Node {
    int data;
    struct Node* next;
};

// Function to create a new node
struct Node* createNode(int value) {
    struct Node* newNode = (struct Node*) malloc(sizeof(struct Node));
    newNode->data = value;
    newNode->next = NULL;
    return newNode;
}

// Function to append a node to the linked list
void append(struct Node** head, int value) {
    struct Node* newNode = createNode(value);
    if (*head == NULL) {
        *head = newNode;
        return;
    }
    struct Node* temp = *head;
    while (temp->next != NULL)
        temp = temp->next;
    temp->next = newNode;
}

// Function to print the linked list
void printList(struct Node* head) {
    struct Node* temp = head;
    while (temp != NULL) {
        printf("%d ", temp->data);
        temp = temp->next;
    }
    printf("\n");
}

// Function to delete the x-th node (1-based index)
struct Node* deleteNode(struct Node* head, int x) {
    if (head == NULL)
        return NULL;

    // Deleting the first node
    if (x == 1) {
        struct Node* temp = head;
        head = head->next;
        free(temp);
        return head;
    }

    struct Node* current = head;
    for (int i = 1; i < x - 1 && current != NULL; i++) {
        current = current->next;
    }

    if (current == NULL || current->next == NULL)
        return head;  // Position out of bounds

    struct Node* temp = current->next;
    current->next = temp->next;
    free(temp);

    return head;
}

int main() {
    struct Node* head = NULL;
    int n, val, x;

    // Input number of nodes
    printf("Enter number of elements: ");
    scanf("%d", &n);

    // Input node values
    printf("Enter %d elements:\n", n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &val);
        append(&head, val);
    }

    // Input position to delete
    printf("Enter position to delete (1-based): ");
    scanf("%d", &x);

    // Delete node
    head = deleteNode(head, x);

    // Print updated list
    printf("Updated linked list:\n");
    printList(head);

    return 0;
}
